### PR TITLE
Dashboard update

### LINKED ISSUE #121: Dashboard options (frontend)
## Requirement
Semester info needs to be added to courses in the backend.

## Description
In the dashboard there could be options to sort the courses into semesters or by dates.

Only show courses of the current semester on dashboard and currently attending part of navbar.

## Acceptance Criteria
- [ ] navbar is adapted
- [ ] dashboard is adapted

### LINKED ISSUE #165: Dashboard make title clickable
## Description
Make the title of courses in the Dashboard clickable and remove the continue button at the bottom.

## Acceptance Criteria
- [ ] title is clickable

### PR DESCRIPTION
**Pull request by SkyDiverStar**
# Description of changes

closes #121 
closes #165 

Could not get 121 to work with yearDivision, so I commented the parts out. Maybe you can find out what went wrong, otherwise I will remove it.

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: app/(dashboard)/student.tsx [hunk 1]
```diff
@@ -34,26 +45,98 @@ export default function StudentPage() {
...
+    setShowAllCourses(!showAllCourses);
+  };
 
   return (
     <main>
       <Typography variant="h1" gutterBottom>
         Dashboard
       </Typography>
+      {/* Sort by Selection */}
+      <Box sx={{ minWidth: 120, maxWidth: 150, float: "right" }}>
```
Comment (mardom1) [inline app/(dashboard)/student.tsx:118]: Better not use float

File: app/(dashboard)/student.tsx [hunk 2]
```diff
@@ -34,26 +45,98 @@ export default function StudentPage() {
...
+        <Typography variant="h6" gutterBottom>
+          Current Year: {currentYear}
+        </Typography>
+      )}
+
       {/* MOCK */}
-      {courses.length === 0 ? (
+      {filteredCourses.length === 0 ? (
         <div className="text-center text-2xl text-slate-400 my-80">
           You have not joined any courses yet. Visit the{" "}
```
Comment (mardom1) [inline app/(dashboard)/student.tsx:141]: "... courses for this year."

File: app/(dashboard)/student.tsx [hunk 3]
```diff
@@ -134,16 +219,105 @@ export default function StudentPage() {
...
+        notFocusesdcourses.length > 0 &&
+        sortBy !== "" &&
+        sortBy !== "title" && (
+          <>
+            <div className="flex flex-col items-center">
+              <Button onClick={handleButton}>Hide All Courses</Button>
+            </div>
+            <div className="grid grid-cols-2 lg:grid-cols-3 gap-8 mt-8">
+              {notFocusesdcourses.map((course) => (
+                <Card variant="outlined" className="h-full" key={course.id}>
```
Comment (mardom1) [inline app/(dashboard)/student.tsx:244]: Please, refactor this into a separate component to avoid duplication

**Review by mardom1 — CHANGES_REQUESTED** [review]
When a course does not have a startYear, it is currently being displayed under "all courses". We should probably fall back to the startDate in this case. It would also be useful to show the year on the course somewhere (as a chip for example). The navbar and dashboard using different logic for filtering the current courses is also a bit confusing in my opinion.

That said, I find the whole startYear redundant anyways, as this is the exact purpose of the the startDate/endDate. Especially as it is not a mandatory field.

File: app/(dashboard)/student.tsx (hunk 1 above)
Comment (SkyDiverStar) [reply app/(dashboard)/student.tsx:118]: What would you suggest? Absolute right, was the only other thing I tried, but that is not useable in this use case.

**Comment by SkyDiverStar** [comment]
Good that you point out the thing with the startYear. Thought I did change it to startDate.year(), might have been delete in one of the undos 😅 
I agree with your opinion. In the beginning I believed it would make things easier, but it does make things harder.
For the navbar it was said that only courses of the current semester are shown, but we don't want to support only semesters, trimesters and quarters. Because the yearDivsion stuff does not work (out commented the parts in the code), I said only courses of the last 7 months are shown. In the dashboard it was just said, to show the courses sorted by year (thinking of improving second array mapping)

File: app/(dashboard)/student.tsx (hunk 2 above)
Comment (SkyDiverStar) [reply app/(dashboard)/student.tsx:141]: Don't understand what you mean?

File: app/(dashboard)/student.tsx (hunk 3 above)
Comment (SkyDiverStar) [reply app/(dashboard)/student.tsx:244]: Meaning the card? If yes, should I do it above as well?

Comment (mardom1) [reply app/(dashboard)/student.tsx:244]: Yes, as I understand the card above is the same as this, so we should not duplicate the code

File: app/(dashboard)/student.tsx (hunk 2 above)
Comment (mardom1) [reply app/(dashboard)/student.tsx:141]: It is possible to see this message while having joined courses (if they are all from previous years). I would change the message slightly to clarify this

File: app/(dashboard)/student.tsx (hunk 1 above)
Comment (mardom1) [reply app/(dashboard)/student.tsx:118]: In most cases where you want to align something, flexbox does the trick

**Comment by SkyDiverStar** [comment]
Chips will come today

**Review by mardom1 — APPROVED** [review]
Looks good, thanks
